src/Simplify.elm

```diff
@@ -6105,6 +6118,51 @@ randomMapCompositionChecks checkInfo =
     wrapperMapCompositionChecks randomGeneratorWrapper checkInfo
 
 
+randomAndThenChecks : CheckInfo -> Maybe (Error {})
+randomAndThenChecks checkInfo =
+    firstThatConstructsJust
+        [ \() -> wrapperAndThenChecks randomGeneratorWrapper checkInfo
+        , \() -> nonEmptiableWrapperAndThenAlwaysChecks randomGeneratorWrapper checkInfo
+        ]
+        ()
+
+
+nonEmptiableWrapperAndThenAlwaysChecks :
+    NonEmptiableProperties (WrapperProperties otherProperties)
+    -> CheckInfo
+    -> Maybe (Error {})
+nonEmptiableWrapperAndThenAlwaysChecks wrapper checkInfo =
+    case AstHelpers.getAlwaysResult checkInfo.lookupTable checkInfo.firstArg of
+        Just alwaysResult ->
+            Just
+                (let
+                    replacementAndFix : { replacementDescription : String, fix : List Fix }
+                    replacementAndFix =
+                        case secondArg checkInfo of
+                            Nothing ->
+                                { replacementDescription = "always with the " ++ wrapper.represents ++ " produced by the function"
+                                , fix =
+                                    -- maybe a bit too clever?
```

Comment:
Why do you think it's too clever?
I'm not 100% confident but I think this is fine (at least with the current way `getAlwaysResult` is implemented.

Reply:
Yeah no it should be fine in practice. I just thought that conceptually you aren't really "returning the given function" but "returning always with the value returned by the given function". They both just accidentally align which is why I added the comment :)

Reply:
Yeah, remove that comment :+1: